sumologic/provider.go

```diff
@@ -134,22 +143,37 @@ func resolveRedirectURL(accessId string, accessKey string) (string, error) {
 func providerConfigure(d *schema.ResourceData) (interface{}, error) {
 	accessId := d.Get("access_id").(string)
 	accessKey := d.Get("access_key").(string)
+	authJwt := d.Get("auth_jwt").(string)
 	environment := d.Get("environment").(string)
 	baseUrl := d.Get("base_url").(string)
 	isInAdminMode := d.Get("admin_mode").(bool)
 
 	msg := ""
-	if accessId == "" {
-		msg = "sumologic provider: access_id should be set;"
-	}
-
-	if accessKey == "" {
-		msg = fmt.Sprintf("%s access_key should be set; ", msg)
+	if authJwt == "" {
+		if accessId == "" || accessKey == "" {
+			msg = "sumologic provider: auth_jwt is not set;"
+		}
+		if accessId == "" {
+			msg = fmt.Sprintf("%s access_id should be set;", msg)
+		}
+		if accessKey == "" {
+			msg = fmt.Sprintf("%s access_key should be set; ", msg)
+		}
+	} else {
+		if accessId != "" || accessKey != "" {
+			msg = "sumologic provider: auth_jwt is set;"
+		}
+		if accessId != "" {
+			msg = fmt.Sprintf("%s access_id should not be set;", msg)
+		}
+		if accessKey != "" {
+			msg = fmt.Sprintf("%s access_key should not be set; ", msg)
+		}
```

Comment:
Given that JWT auth is for our internal use case, should we ignore accessId/Key if JWT  auth is set?

Reply:
Yes, with changing access id/key back to `Required`, this section is removed.